src/macros.rs

```diff
@@ -131,13 +131,12 @@ macro_rules! implement_vertex {
                                 offset
                             },
                             {
-                                fn attr_type_of_val<T: $crate::vertex::Attribute>(_: &T)
+                                fn attr_type_of_val<T: $crate::vertex::Attribute>(_: Option<&T>)
                                     -> $crate::vertex::AttributeType
                                 {
                                     <T as $crate::vertex::Attribute>::get_type()
                                 }
-                                let dummy: &$struct_name = unsafe { ::std::mem::transmute(0usize) };
-                                attr_type_of_val(&dummy.$field_name)
+                                attr_type_of_val(None::<&$struct_name>.map(|v| v.$field_name))
```

Comment:
Whatever is going on here looks clever... might be a good idea to add a comment explaining it.

The `map` closure is dead code, right? Just present to make the type checker do stuff?

Reply:
You can look at #1776 for context

Reply:
Sure, but that doesn't help for someone reading the code in half a year.

Reply:
It's not *that* clever to me. It's a method to obtain the type of the `$field_name` field and call get_type on it without having to construct an instance of the struct.

Reply:
Also it's safe code now. But I'll add a comment regardless.